lib/os.js

```diff
@@ -190,17 +185,33 @@ function getCIDR({ address, netmask, family }) {
 }
 
 function networkInterfaces() {
-  const interfaceAddresses = getInterfaceAddresses();
+  const data = getInterfaceAddresses();
+  const result = Object.create(null);
 
-  const keys = Object.keys(interfaceAddresses);
-  for (var i = 0; i < keys.length; i++) {
-    const arr = interfaceAddresses[keys[i]];
-    for (var j = 0; j < arr.length; j++) {
-      arr[j].cidr = getCIDR(arr[j]);
-    }
+  if (data === undefined)
+    return result;
+  for (var i = 0; i < data.length; i += 7) {
+    const name = data[i];
+    const entry = {
+      address: data[i + 1],
+      netmask: data[i + 2],
+      family: data[i + 3],
+      mac: data[i + 4],
+      internal: data[i + 5],
+      cidr: getCIDR(data[i + 1], data[i + 2], data[i + 3])
+    };
+    const scopeid = data[i + 6];
+    if (scopeid !== -1)
+      entry.scopeid = scopeid;
+
+    const existing = result[name];
+    if (existing !== undefined)
+      existing.push(entry);
+    else
+      result[name] = [entry];
   }
```

Comment:
Can you explain the perf benefit for `networkInterfaces()`.

Reply:
What do you mean?

Reply:
At first glance it looks like a lot of unrolled code with an extra object being created. Could you please step me through the optimization being done in `networkInterfaces()`.

Reply:
If you drop the rework to `networkInterfaces()` do you still see perf wins?

Reply:
It's more or less moving the object creation from C++ to JS land.